nat-handler.py

```diff
@@ -94,17 +102,30 @@ def current_az(self):
     def current_region(self):
         return self.current_az[:-1]
 
+    @property
+    def primary_interface_id(self):
+        ec2 = connect_to_ec2(self.current_region)
+        instance = ec2.get_only_instances(instance_ids=[self.current_instance_id])[0]
+        interfaces = instance.interfaces
+        interface_id = [i.id for i in interfaces if i.id != self.secondary_interface_id]
+        return interface_id[0]
+
+    @property
+    def secondary_interface_id(self):
+        return self.config.secondary_interface_id(self.current_az)
+
     def take_route(self, az):
         route_table_id = self.config.route_table_id(az)
         vpc = connect_to_vpc(self.current_region)
-        vpc.replace_route(route_table_id, '0.0.0.0/0', instance_id=self.current_instance_id)
+        vpc.replace_route(route_table_id, '0.0.0.0/0', interface_id=self.primary_interface_id)
 
     def take_elastic_ip(self, az):
+        interface_id = self.primary_interface_id if az == self.current_az else self.secondary_interface_id
```

Comment:
Your line of thought is correct, if there are exactly three availability zones. This is the case in eu-west-1, but not in us-east-1. There I believe they have five availability zones.

For our purposes it matters not, but you might want to add a comment to the README so other users are not suprised by the unclear behaviour.